src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs

```diff
@@ -1081,46 +1121,62 @@ public static void Equal(Half expected, Half actual, Half variance)
             {
                 throw EqualException.ForMismatchedValues(ToStringPadded(expected), ToStringPadded(actual));
             }
+        }
+#endif
 
-            static unsafe bool IsNegativeZero(Half value)
+        /// <summary>Verifies that two <see cref="double"/> values's binary representations are identical.</summary>
+        /// <param name="expected">The expected value</param>
+        /// <param name="actual">The value to be compared against</param>
+        /// <exception cref="EqualException">Thrown when the representations are not identical</exception>
+        public static void Equal(double expected, double actual)
+        {
+            if (BitConverter.DoubleToInt64Bits(expected) == BitConverter.DoubleToInt64Bits(actual))
             {
-                return (*(ushort*)(&value)) == 0x8000;
+                return;
             }
 
-            static unsafe bool IsPositiveZero(Half value)
+            throw EqualException.ForMismatchedValues(ToStringPadded(expected), ToStringPadded(actual));
+        }
+
+        /// <summary>Verifies that two <see cref="float"/> values's binary representations are identical.</summary>
+        /// <param name="expected">The expected value</param>
+        /// <param name="actual">The value to be compared against</param>
+        /// <exception cref="EqualException">Thrown when the representations are not identical</exception>
+        public static void Equal(float expected, float actual)
+        {
+            if (BitConverter.SingleToInt32Bits(expected) == BitConverter.SingleToInt32Bits(actual))
             {
-                return (*(ushort*)(&value)) == 0x0000;
+                return;
             }
 
-            // We have a custom ToString here to ensure that edge cases (specifically +-0.0,
-            // but also NaN and +-infinity) are correctly and consistently represented.
-            static string ToStringPadded(Half value)
+            if (PlatformDetection.IsRiscV64Process && float.IsNaN(expected) && float.IsNaN(actual))
```

Comment:
I'd expect this same check to exist for `double` as well, for overall consistency

Reply:
Done, however there was no problem with double due to float->half and double->half cast algorithms are different.